src/resource/ResourceManager.h

```diff
@@ -18,13 +20,23 @@
 namespace Ship {
 struct File;
 
+struct ResourceFilter {
+    ResourceFilter(const std::list<std::string> includeMasks, const std::list<std::string> excludeMasks,
```

Comment:
Can we not make the lists that are passed into the constructors here references and then copy into new containers on construct?

Reply:
Not that I could figure out, that was the very issue that was causing that problem I was talking about on Discord last night where it would end up with garbage data in the containers while iterating through them in `ListFiles`.

Reply:
Changing them to vector references didn't work, either, though I would have thought it would from the research I did on the issue saying that lists can have reference passing issues with non-contiguous memory, etc.

Reply:
What about const pointers instead? Raw pointers.

Reply:
LUS would have to copy anyway... I don't think we want LUS relying on memory blocks allocated by the caller.

In this setup the optimizer _should_ be able to see the double copy and chop one out. I can see why it's extra rough in debug build.

Reply:
Somehow, now, that doesn't work either. This is an example of what's going on with it, when this is supposed to be a list with a size of 4
![image](https://github.com/user-attachments/assets/357b40de-de6f-4b57-a200-00ae7f4dab8b)

Reply:
Ah, I see now what you were talking about before, and I realize I didn't copy the `ResourceIdentifier` precedent exactly. Yes, that works just fine. So if they're not references in the arguments, does that mean they would be copied twice or something? What's the benefit to having references passed in but copied to internal storage?

Reply:
Yeah it will copy once into the function, and then it will copy a second time to make one inside of LUS. This is the literal code. I _hope_ release optimizations handle this.